### PR TITLE
Edits/topic 2 1

### PR DESCRIPTION
**Pull request by gigamonkey**
Okay, this is my first edit of some content that actually matters. Some key goals I had with this change:

- Deemphasize the idea of "object in programs map to objects in the real world" as that tends to lead to weird OO designs.
- My hobby horse of being very precise about the distinction between values and variables.
- Introducing the idea of a class as defining a type and tying that back to what the student has already learned about types in Unit 1

I'll keep saying this for a while yet: I'm totally fine if you don't want any of these changes. Or even if you just don't want to deal with a barrage of changes from me. Feel free to ask me to cool it if this isn't actually helpful.

### CODE REVIEW

File: _sources/Unit2-Using-Objects/topic-2-1-objects-intro-turtles.rst [hunk 1]
```diff
@@ -44,14 +44,39 @@
 Objects - Instances of Classes
 ===============================
 
+Java is an **object-oriented programming** language. That means that after we’ve
+exhausted the possibilities of primitive types like ``int``, ``double``, and
+``boolean``, one of the primary ways of organizing our programs is in terms of
```
Comment (bhoffman0) [inline _sources/Unit2-Using-Objects/topic-2-1-objects-intro-turtles.rst:49]: Not sure about this sentence  and its two not so related parts: "That means that after we’ve
exhausted the possibilities of primitive types like ``int``, ``double``, and
``boolean``, one of the primary ways of organizing our programs is in terms of
**objects**."

**Comment by bhoffman0** [comment]
I like these 3 bulleted points you make!

File: _sources/Unit2-Using-Objects/topic-2-1-objects-intro-turtles.rst [hunk 2]
```diff
@@ -60,9 +85,21 @@ You can think of a class like a blueprint or a cookie cutter.  It is used to cre
...
-
-The following picture has lots of cats (objects of the type cat).  They are all different, but they share the same attributes and behaviors that make up a cat. They are all **instances** of a cat with different values for their attributes. Name some of the attributes and behaviors of the cats below. For example, the color (*attribute*) of the first cat is black (*attribute value*) and it is playing (*behavior*).
+You can also think of a class defining a new type. Just like you use ``int`` to
+declare variables that can hold an whole number value, you can use ``Turtle`` to
+declare a variable whose value has to be an instance of the ``Turtle`` class.
+And just like the Java compiler will only let you do things with the values of
+``int`` variables that make sense (like adding and multiplying them), it will
+only let you do thigs with values of a ``Turtle`` variable that make sense to do
```
Comment (bhoffman0) [inline _sources/Unit2-Using-Objects/topic-2-1-objects-intro-turtles.rst:93]: thigs -> things

**Review by bhoffman0 — APPROVED** [review]

File: _sources/Unit2-Using-Objects/topic-2-1-objects-intro-turtles.rst (hunk 2 above)
Comment (gigamonkey) [inline _sources/Unit2-Using-Objects/topic-2-1-objects-intro-turtles.rst:93]:
```suggestion
only let you do things with values of a ``Turtle`` variable that make sense to do
```

**Review by bhoffman0 — APPROVED** [review]

**Pull request merged**